Diagnostics/HealthChecker/Features/Invoke-VulnerabilityReport.ps1

```diff
@@ -0,0 +1,116 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\Helpers\Get-ExportedHealthCheckerFiles.ps1
+# This function is used to collect the data from the entire environment for vulnerabilities.
+# This will still use the entire data collection process that is handled within the main HC data collection
+# However, it will not display it on the screen and just analyze the data and pull out the data that we want.
+function Invoke-VulnerabilityReport {
+
+    Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+    $stopWatch = [System.Diagnostics.Stopwatch]::StartNew()
+    Set-ADServerSettings -ViewEntireForest $true
+    $exchangeServers = @(Get-ExchangeServer)
+    Write-Verbose "Took $($stopWatch.Elapsed.TotalSeconds) seconds to run Get-ExchangeServer"
+    Write-Verbose "Found $($exchangeServers.Count) Exchange Servers"
+
+    $healthCheckerData = New-Object System.Collections.Generic.List[object]
+    $serverNames = New-Object System.Collections.Generic.List[string]
+    $exchangeServers | ForEach-Object { $serverNames.Add($_.Name.ToLower()) }
+
+    # By default, always try to import the local data and see if we can use it.
+    # This is determined by the files being the same version as the current version of the script.
+    $importData = Get-ExportedHealthCheckerFiles -Directory $XMLDirectoryPath
+
+    if ($null -ne $importData) {
+        $importData |
+            Where-Object { $_.HealthCheckerExchangeServer.HealthCheckerVersion -eq $BuildVersion } |
```

Comment:
Can we extract the `DateTime` string from the xml file (`HealthChecker-e2k16-2-20221228152212.xml`) and compare it against the current DateTime and decide then whether we use the existing xml files or not (e.g., if they are > 1 week old, re-collect the data to avoid using outdated files)?

Reply:
Or are we good by using the HC version as an indicator?

Reply:
HC version is likely the best indicator at this time.